File: tests/fixtures/edgecase_repo/php/module1.php
<?php
namespace Example\Gen;

use Example\Gen\Base1;
use Example\Gen\Handler1 as H1;

class Base1
{
    protected $name;
}

interface Handler1
{
    public function handle($payload);
}

class Service1 extends Base1 implements Handler1
{
    public function handle($payload): bool
    {
        $this->validate($payload);
        return true;
    }

    private function validate($payload): void
    {
    }
}

function helper_1($value) {
    return $value;
}

File: tests/fixtures/edgecase_repo/cpp/module1.cpp
#include <string>
#include "module0.hpp"

namespace gen {

class Base1 {
public:
    std::string name;
};

class Service1 : public Base1 {
public:
    bool handle(const std::string &payload) {
        this->validate(payload);
        return true;
    }

    void validate(const std::string &payload) {}
};

int helper_1(int value) { return value; }

}  // namespace gen
